revolution_pi/gpio_pin.go

```diff
@@ -145,56 +233,66 @@ func (pin *gpioPin) PWMFreq(ctx context.Context, extra map[string]interface{}) (
 	if !pin.initialized {
 		return 0, errors.New("pin not initialized")
 	}
+	if !pin.isOutputPWM() && !pin.isOutputWord() {
+		return 0, fmt.Errorf("cannot get PWM Frequency, Pin %s is not a PWM pin", pin.Name)
+	}
 
 	b := make([]byte, 1)
-	n, err := pin.ControlChip.fileHandle.ReadAt(b, 112)
+	// all PWM pins use the same PWM frequency
+	n, err := pin.ControlChip.fileHandle.ReadAt(b, int64(pin.inputOffset+outputPWMFrequencyOffset))
 	if err != nil {
 		return 0, err
 	}
 	if n != 1 {
 		return 0, errors.New("unable to read PWM Frequency")
 	}
-	pin.ControlChip.logger.Infof("Current frequency: %#v", b)
+	pin.ControlChip.logger.Debugf("Current frequency step size: %#d", b)
 
-	return 0, nil
+	return stepSizeToFreq(b), nil
+}
+
+func stepSizeToFreq(step []byte) uint {
+	// b only has one byte
+	switch step[0] {
+	case 1:
+		return 40
+	case 2:
+		return 80
+	case 4:
+		return 160
+	case 5:
+		return 200
+	case 10:
+		return 400
+	}
+	return 0
 }
 
-// SetPWMFreq sets the given pin to the given PWM frequency. For Raspberry Pis,
-// 0 will use a default PWM frequency of 800.
+// SetPWMFreq sets the given pin to the given PWM frequency. For the Rev-Pi this must be configured in PiCtory
 func (pin *gpioPin) SetPWMFreq(ctx context.Context, freqHz uint, extra map[string]interface{}) error {
 	if !pin.initialized {
 		return errors.New("pin not initialized")
 	}
 
-	return errors.New("not supported")
+	return errors.New("PWM Frequency must be set in PiCtory")
```

Comment:
most configuration parameters have to be handled in PiCtory. We could attempt to regenerate an entire config file and have the module restart the entire board if we really wanted to configure these in the module, but I believe we shouldn't reinvent the wheel. Can discuss this more in the scope